design-system/atoms/input/adapters/default/default.adapter.tsx

```diff
@@ -0,0 +1,230 @@
+        return true;
+      }
+    }
+
+    return false;
+  }, [button]);
+
+  const hasEndContent = endIcon || endContent || isInnerButton;
+
+  if (!hasEndContent) {
+    return null;
+  }
+
+  if (isInnerButton) {
+    return (
+      <div className={"px-1"}>
+        <Button {...button} isDisabled={variants?.isDisabled} />
+      </div>
+    );
+  }
+
+  return (
+    <div className={cn(slots.contentWrapper())}>
+      <div className={"flex flex-row items-center justify-end gap-1"}>
+        {endContent}
+        {endIcon ? <Icon size={"sm"} {...endIcon} /> : null}
+      </div>
+    </div>
+  );
+}
+
+export const InputDefaultAdapter = forwardRef(function InputDefaultAdapter(
+  {
+    id,
+    name,
+    classNames,
+    isError,
+    isDisabled,
+    value,
+    onChange,
+    startContent,
+    startIcon,
+    avatar,
+    endContent,
+    info,
+    error,
+    description,
+    endIcon,
+    label,
+    placeholder,
+    canInteract,
+    button,
+    attr = {},
+    size = defaultSize,
+    isFocused: _isFocused,
+  }: InputPort,
+  ref: ForwardedRef<HTMLInputElement>
+) {
+  const [isFocused, setIsFocused] = useState(false);
+  const isOuterButton = useMemo(() => {
+    if (button) {
+      if (isButtonSolid(button)) {
+        return button.variant === "secondary";
+      }
+    }
+
+    return false;
+  }, [button]);
+
+  const variants = { isFocused: isFocused || _isFocused, size, isDisabled, isError, asOuterElement: isOuterButton };
+  const slots = InputDefaultVariants(variants);
+
+  function onFocus() {
+    setIsFocused(true);
+  }
+
+  function onBlur() {
+    setIsFocused(false);
+  }
+
+  function handleChanges(e: ChangeEvent<HTMLInputElement>) {
+    if (onChange) {
+      onChange(e);
+    }
+  }
+
+  return (
+    <div className={cn(slots.container(), classNames?.container)} {...attr}>
+      <LabelsContainer label={label} description={description} classNames={classNames} />
+      <div className={cn(slots.wrapper())} {...attr}>
+        <div className={cn(slots.base(), classNames?.base)} {...attr}>
+          <StartContent startContent={startContent} startIcon={startIcon} avatar={avatar} {...variants} />
+          <div className={cn(slots.inputWrapper())}>
+            <input
+              name={name}
+              id={id}
+              ref={ref}
+              className={cn(slots.input(), classNames?.input)}
+              onFocus={onFocus}
+              onBlur={onBlur}
+              value={value}
+              placeholder={placeholder}
+              disabled={isDisabled || canInteract}
+              onChange={handleChanges}
+            />
+          </div>
+          <EndContent endContent={endContent} endIcon={endIcon} button={button} {...variants} />
+        </div>
+        {isOuterButton && button ? (
+          <Button
+            {...button}
+            size={size}
+            classNames={{ base: "rounded-l-none !border-l-0 h-auto" }}
+            isDisabled={isDisabled}
+          />
```

Comment:
I don't see this in the designs ?

Reply:
<img width="254" alt="Capture d’écran 2024-09-05 à 17 04 44" src="https://github.com/user-attachments/assets/eec4e206-ade2-4a2e-8a2c-6952c34f7597">